fuzzers/libafl_atheris/src/lib.rs

```diff
@@ -218,6 +217,7 @@ pub extern "C" fn LLVMFuzzerRunDriver(
 
         // Setup a tracing stage in which we log comparisons
         let tracing = TracingStage::new(InProcessExecutor::new(
+            tuple_list!(),
```

Comment:
Can we have `new` be with empty hooks, so the API doesn't change, and then have `::with_hooks` for the new API?

Reply:
case in point, there is not a single example that uses hooks, so it seems to be _extremely_ nieche

Reply:
> Can we have new be with empty hooks, so the API doesn't change, and then have ::with_hooks for the new API?
this is the hard part...

At first I also wanted to do like this.
the problem is the type `HT` of the InProcessExecutor needs to be decided at compile time.
therefore I cannot make a new constructor that initialized with the empty list (because () != HT)

Reply:
Then `impl Ht for ()`?
Or use a dummy type

Reply:
You should be able to use `tuple_list_type!()` or just `(())` as type

Reply:
> Then impl Ht for ()?

it is already implemented.
```
error[E0308]: mismatched types
   --> libafl/src/executors/inprocess.rs:235:13
    |
211 | impl<H, HB, HT, OT, S> GenericInProcessExecutor<H, HB, HT, OT, S>
    |             -- expected this type parameter
...
234 |         Self::with_timeout(
    |         ------------------ arguments to this function are incorrect
235 |             tuple_list!(),
    |             ^^^^^^^^^^^^^ expected type parameter `HT`, found `()`
    |
    = note: expected type parameter `HT`
                    found unit type `()`
```

The problem is the type mismatch between HT and ()
and it needs to be given from the outside of this struct

Reply:
I have a solution for this

Reply:
fixed